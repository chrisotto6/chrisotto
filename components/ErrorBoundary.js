import * as Sentry from '@sentry/nextjs'
import * as React from 'react'

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props)
    this.state = { hasError: false }
  }
  static getDerivedStateFromError(error) {
    return { hasError: true }
  }
  componentDidCatch(error, errorInfo) {
    // You can use your own error logging service here
    Sentry.captureException(errorInfo)
  }
  render() {
    // Check if the error is thrown
    if (this.state.hasError) {
      // You can render any custom fallback UI
      return (
        <div>
          <h2>Oops, there is an error!</h2>
          <button type="button" onClick={() => this.setState({ hasError: false })}>
            Try again?
          </button>
        </div>
      )
    }
    return this.props.children
  }
}

export default ErrorBoundary
