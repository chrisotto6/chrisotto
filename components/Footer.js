import Link from 'next/link'
import siteMetadata from '@/data/siteMetadata'
import { FaEnvelope, FaGithub, FaLinkedin, FaTwitter } from 'react-icons/fa'

export default function Footer() {
  return (
    <footer>
      <div className="mt-16 flex flex-col items-center">
        <div className="mb-5 flex space-x-2 text-sm text-gray-500 dark:text-gray-400">
          <Link href="/">{siteMetadata.title}</Link>
          <div>{` • `}</div>
          <div>{`© ${new Date().getFullYear()}`}</div>
        </div>
        <div className="mb-3 flex space-x-8">
          <Link href={`mailto:${siteMetadata.email}`}>
            <FaEnvelope size={30} />
          </Link>
          <Link href={siteMetadata.github}>
            <FaGithub size={30} />
          </Link>
          <Link href={siteMetadata.linkedin}>
            <FaLinkedin size={30} />
          </Link>
          <Link href={siteMetadata.twitter}>
            <FaTwitter size={30} />
          </Link>
        </div>
      </div>
    </footer>
  )
}
