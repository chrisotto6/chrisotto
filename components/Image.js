import NextImage from 'next/image'

// eslint-disable-next-line jsx-a11y/alt-text
const Image = ({ ...rest }) => (
  <NextImage
    {...rest}
    style={{
      maxWidth: '100%',
      height: 'auto',
    }}
    sizes="100vw"
  />
)

export default Image
