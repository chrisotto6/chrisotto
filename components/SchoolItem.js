import kebabCase from '@/lib/utils/kebabCase'

const SchoolItem = ({ schoolName, degree, gpa, awards }) => {
  return (
    <div data-testid={`school-${kebabCase(schoolName)}`}>
      <h3>{schoolName}</h3>
      <h4>{degree}</h4>
      <h5>{`${gpa} - ${awards}`}</h5>
    </div>
  )
}

export default SchoolItem
