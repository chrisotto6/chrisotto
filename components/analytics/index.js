import GA from './GoogleAnalytics'
import siteMetadata from '@/data/siteMetadata'

const isProduction = process.env.NODE_ENV === 'production'

const Analytics = () => {
  return <>{isProduction && siteMetadata.analytics.googleAnalyticsId && <GA />}</>
}

export default Analytics
