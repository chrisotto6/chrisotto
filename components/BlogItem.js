import Image from './Image'
import Link from 'next/link'
import Tag from '@/components/Tag'
import formatDate from '@/lib/utils/formatDate'

const BlogItem = ({ slug, date, title, summary, tags, image, readingTime }) => (
  <article
    // prettier-ignore
    className="display:block w-full gap-6 space-y-2 rounded-xl xl:grid xl:grid-cols-2 xl:items-start dark:hover:bg-sky-700/40 hover:bg-gray-100"
    data-testid="list-blog-item"
  >
    <Link href={`/blog/${slug}`} className="text-gray-900 dark:text-gray-100">
      <Image className="w-full rounded-xl" alt={title} src={image} width={450} height={250} />
    </Link>
    <div className="space-y-6">
      <div className="space-y-5 xl:col-span-3">
        <Link href={`/blog/${slug}`} className="text-gray-900 dark:text-gray-100">
          <div>
            <h2 className="text-2xl font-bold leading-8 tracking-tight text-gray-900 dark:text-gray-100">
              {title}
            </h2>
            <dl>
              <dt className="sr-only">Published on</dt>
              <dd className="inline-block text-base font-medium leading-6 text-gray-500 dark:text-gray-400">
                <time dateTime={date}>{formatDate(date)}</time>
                <p>{readingTime.text}</p>
              </dd>
            </dl>
          </div>
          <div className="prose max-w-none text-gray-500 dark:text-gray-400">{summary}</div>
        </Link>
        <div className="flex flex-wrap" data-testid="tag-container">
          {tags.map((tag) => (
            <Tag key={tag} text={tag} />
          ))}
        </div>
      </div>
    </div>
  </article>
)

export default BlogItem
