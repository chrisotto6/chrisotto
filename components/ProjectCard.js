import Image from './Image'
import Link from 'next/link'
import { FaGithub, FaExternalLinkSquareAlt } from 'react-icons/fa'

const ProjectCard = ({ title, description, img, githubUrl, projectUrl, tech, dataTestid }) => (
  <div
    className="max-w-xs overflow-hidden rounded-xl shadow-lg dark:bg-gray-600 dark:shadow-cyan-600/50"
    data-testid={dataTestid}
  >
    <Image className="w-full" alt={title} src={img} width={544} height={306} />
    <div className="px-6 py-4">
      <div className="mb-2 text-xl font-bold">{title}</div>
      <p className="text-base text-gray-500 dark:text-gray-100">{description}</p>
    </div>
    <div className="flex px-6 pt-3 pb-2">
      {tech.map((t) => (
        <span
          key={t}
          className="mr-2 mb-2 inline-block rounded-full bg-gray-200 px-3 py-1 text-sm font-semibold text-gray-500 dark:bg-cyan-600 dark:text-gray-100"
        >
          {t}
        </span>
      ))}
    </div>
    <div className="mb-3 flex space-x-4 px-6 pt-3 pb-3">
      <Link href={githubUrl}>
        <FaGithub size={30} />
      </Link>
      <Link href={projectUrl}>
        <FaExternalLinkSquareAlt size={30} />
      </Link>
    </div>
  </div>
)

export default ProjectCard
