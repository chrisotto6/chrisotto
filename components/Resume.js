import schoolData from '@/data/schoolData'
import volunteerData from '@/data/volunteerData'
import workData from '@/data/workData'
import SchoolItem from '@/components/SchoolItem'
import VolunteerItem from '@/components/VolunteerItem'
import WorkItem from '@/components/WorkItem'
import { title } from '@/data/siteMetadata'

const Resume = () => {
  return (
    <article>
      <h2>Experience</h2>
      {workData.map((job) => (
        <WorkItem key={job} company={job.company} role={job.role} />
      ))}
      <h2>Education</h2>
      {schoolData.map((school) => (
        <SchoolItem
          key={school.schoolName}
          degree={school.degree}
          gpa={school.gpa}
          awards={school.awards}
        />
      ))}
      <h2>Volunteering</h2>
      {volunteerData.map((vol) => [
        <VolunteerItem
          key={vol.company}
          company={vol.company}
          title={vol.title}
          summary={vol.summary}
          yearStart={vol.yearStart}
          yearEnd={vol.yearEnd}
        />,
      ])}
    </article>
  )
}

export default Resume
