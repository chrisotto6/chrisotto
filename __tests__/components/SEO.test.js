import { render } from '@testing-library/react'
import { CommonSEO, PageSEO, BlogSEO, TagSEO } from '@/components/SEO'
import * as nextRouter from 'next/router'

nextRouter.useRouter = jest.fn()
nextRouter.useRouter.mockImplementation(() => ({ route: '/' }))

it('renders CommonSEO unchanged', () => {
  const { container } = render(
    <CommonSEO
      title="test"
      description="test"
      ogType="test"
      ogImage="test"
      twImage="test"
      canonicalUrl="test"
    />
  )
  expect(container).toMatchSnapshot()
})

it('renders CommonSEO with ogImage Array unchanged', () => {
  const { container } = render(
    <CommonSEO
      title="test"
      description="test"
      ogType="test"
      ogImage={['', '']}
      twImage="test"
      canonicalUrl="test"
    />
  )
  expect(container).toMatchSnapshot()
})

it('renders PageSEO unchanged', () => {
  const { container } = render(<PageSEO title="test" description="test" />)
  expect(container).toMatchSnapshot()
})

it('renders TagSEO unchanged', () => {
  const { container } = render(<TagSEO title="test" description="test" />)
  expect(container).toMatchSnapshot()
})

it('renders BlogSEO unchanged', () => {
  const { container } = render(
    <BlogSEO
      authorDetails={[
        {
          readingTime: { text: '1 min read', minutes: 0.18, time: 10800, words: 36 },
          slug: ['default'],
          fileName: 'chris.md',
          name: 'Chris Otto',
          avatar: '/static/images/avatar.jpg',
          occupation: 'Manager Software Engineering',
          company: 'Northwestern Mutual',
          email: 'chris@chrisotto.dev',
          twitter: 'https://twitter.com/chris_otto6',
          linkedin: 'https://www.linkedin.com/in/ottochristopher/',
          github: 'https://github.com/chrisotto6',
          date: null,
        },
      ]}
      title="test"
      summary="test"
      date="2021-02-19"
      lastmod="2021-02-19"
      url="test"
      images={['test']}
      canonicalUrl="test"
    />
  )
  expect(container).toMatchSnapshot()
})

it('renders BlogSEO without authorDetails and empty images array unchanged', () => {
  const { container } = render(
    <BlogSEO
      title="test"
      summary="test"
      date="2021-02-19"
      lastmod="2021-02-19"
      url="test"
      images={[]}
      canonicalUrl="test"
    />
  )
  expect(container).toMatchSnapshot()
})

it('renders BlogSEO without authorDetails and empty images array unchanged', () => {
  const { container } = render(
    <BlogSEO
      title="test"
      summary="test"
      date="2021-02-19"
      lastmod="2021-02-19"
      url="test"
      images="test"
      canonicalUrl="test"
    />
  )
  expect(container).toMatchSnapshot()
})
