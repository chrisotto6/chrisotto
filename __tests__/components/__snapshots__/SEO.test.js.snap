// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BlogSEO unchanged 1`] = `<div />`;

exports[`renders BlogSEO without authorDetails and empty images array unchanged 1`] = `<div />`;

exports[`renders BlogSEO without authorDetails and empty images array unchanged 2`] = `<div />`;

exports[`renders CommonSEO unchanged 1`] = `<div />`;

exports[`renders CommonSEO with ogImage Array unchanged 1`] = `<div />`;

exports[`renders PageSEO unchanged 1`] = `<div />`;

exports[`renders TagSEO unchanged 1`] = `<div />`;
